.github/workflows/release-code-freeze.yml

```diff
@@ -0,0 +1,145 @@
+name: "Release - Code freeze"
+
+# This action will run according to the cron schedule or when triggered manually
+on:
+  schedule:
+    - cron: '0 12 * * 0' # Run at 1200 UTC on Sundays.
+  workflow_dispatch:
+    inputs:
+      skipSlackPing:
+        description: "If true, the Slack ping will be skipped"
+        type: boolean
+        required: false
+        default: false
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  check-code-freeze:
+    name: "Check that today is the day of the code freeze"
+    runs-on: ubuntu-20.04
+    outputs:
+      freeze: ${{ steps.check-freeze.outputs.FREEZE }}
+      releaseVersion: ${{ steps.next-version.outputs.RELEASE_VERSION }}
+      releaseDate: ${{ steps.next-version.outputs.RELEASE_PLANNED_DATE }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: 'develop'
+    
+      - name: Get current version
+        id: current-version
+        run: |
+          VERSION=$(jq '.version' package.json -r)
+          echo "Current version found: $VERSION" >> $GITHUB_STEP_SUMMARY
+          echo "VERSION=$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.4'
+          tools: composer
+          coverage: none
+
+      - name: Get the next version
+        id: next-version
+        shell: php {0}
+        env:
+          CURRENT_VERSION: ${{ steps.current-version.outputs.VERSION }}
+        run: |
+          <?php
+          
+          $latest_version_with_release = getenv( 'CURRENT_VERSION' );
+          if ( empty( $latest_version_with_release ) ) {
+            echo "::error::Unable to get the latest released version";
+            exit( 1 );
+          }
+          
+          // Because we go from 5.9 to 6.0, we can get the next major_minor by adding 0.1 and formatting appropriately.
+          $latest_version_as_float = (float) $latest_version_with_release;
+          $branch_major_minor      = number_format( $latest_version_as_float + 0.1, 1 );
+          $next_version            = $branch_major_minor.0;
+          echo "::set-output name=RELEASE_VERSION::$next_version" . PHP_EOL;
+
+          $next_wednesday = date('Y-n-d', strtotime('next Wednesday'));
+          echo "::set-output name=RELEASE_PLANNED_DATE::$next_wednesday" . PHP_EOL;
```

Comment:
`set-output` command is being repaced with new environment files and this command will stop working from June 2023. It be best if we can replace them to the new format - https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Reply:
After discussing with @jbordonado this will be handled as a separate PR, since the implementation is inside PHP.